src/cart/src/cart/crt_rpc.c

```diff
@@ -62,6 +64,36 @@ crt_hdlr_ctl_fi_toggle(crt_rpc_t *rpc_req)
 		D_ERROR("crt_reply_send() failed. rc: %d\n", rc);
 }
 
+void
+crt_hdlr_ctl_log_add_msg(crt_rpc_t *rpc_req)
+{
+	struct crt_ctl_log_add_msg_in	*in_args;
+	struct crt_ctl_log_add_msg_out	*out_args;
+	int				rc = 0;
+
+	in_args = crt_req_get(rpc_req);
+	out_args = crt_reply_get(rpc_req);
+
+	if (in_args->log_msg == NULL) {
+		D_ERROR("Empty log message\n");
+		rc = -DER_INVAL;
+	}
+
+	if (strlen(in_args->log_msg) > CRT_CTL_MAX_LOG_MSG_SIZE) {
+		D_ERROR("Log message exceeded %d chars\n",
+			CRT_CTL_MAX_LOG_MSG_SIZE);
+		rc = -DER_OVERFLOW;
```

Comment:
Should this be a client side check instead?   I think we can still print it but just truncate it

"%*s, CRT_CTL_MAX_LOG_MSG_SIZE, in_arg->log_msg);

should probably also use strnlen for the check

Reply:
Since it is possible for someone to issue rpc on their own without using client, the check needs to be on the server side